src/shipping/consignment-action-creator.ts

```diff
@@ -287,29 +323,52 @@ export default class ConsignmentActionCreator {
         };
     }
 
-    private _combineLineItems(
-        consignment: ConsignmentAssignmentRequestBody,
-        existingConsignment?: Consignment,
+    private _removeLineItems(
+        lineItems: ConsignmentLineItem[],
+        consignment: Consignment,
+        cart?: Cart
+    ): ConsignmentLineItem[] {
+        if (!cart) {
+            throw new MissingDataError(MissingDataErrorType.MissingCart);
+        }
+
+        return this._hydrateLineItems(consignment.lineItemIds, cart).map(existingItem => {
+            const sharedItem = lineItems.find(lineItem => lineItem.itemId === existingItem.itemId);
+
+            return {
+                ...existingItem,
+                quantity: existingItem.quantity - (sharedItem ? sharedItem.quantity : 0),
+            };
+        }).filter(lineItem => lineItem.quantity > 0);
+    }
+
+    private _addLineItems(
+        lineItems: ConsignmentLineItem[],
+        consignment?: Consignment,
         cart?: Cart
     ): ConsignmentLineItem[] {
-        if (!existingConsignment) {
-            return consignment.lineItems;
+        if (!consignment) {
+            return lineItems;
         }
 
         if (!cart) {
             throw new MissingDataError(MissingDataErrorType.MissingCart);
         }
 
-        const existingLineItems = map(existingConsignment.lineItemIds, itemId => {
+        return lineItems
+            .concat(this._hydrateLineItems(consignment.lineItemIds, cart))
+            .filter(lineItem => lineItem.quantity > 0);
+    }
+
+    private _hydrateLineItems(lineItemIds: string[], cart: Cart): ConsignmentLineItem[] {
+        return map(lineItemIds, itemId => {
```

Comment:
🍹I guess if this is a normal array, there's no need for lodash...

```suggestion
        return lineItemIds.map(itemId => {
```